amazonec2/tests/test.bal

```diff
@@ -75,39 +76,42 @@ function testRunInstances() {
         testInstanceIds = arr.map(function (EC2Instance inst) returns (string) {
                 return inst.id;
             });
+        testImageIds = arr.map(function (EC2Instance inst) returns (string) {
+            return inst.imageId;
+        });
         testZoneName = insts[0].zone;
-        test:assertNotEquals(testInstanceIds[0], (), msg = "Failed to run the instances");
+        test:assertEquals(testImageId, testImageIds[0], msg = "Failed to run the instances");
     }
 }
 
 @test:Config {
     dependsOn: ["testRunInstances"]
 }
 function testDescribeInstances() {
-    runtime:sleep(90000);
+    runtime:sleep(50000);
```

Comment:
Why do we need this sleep? Isn't running guaranteed once `testRunInstances()` passes?

Reply:
Yes we need, because after launching the instance it takes some times to become available state.

Reply:
Isn’t there a way to check whether an instance is up and running, instead of having to put random sleeps? Maybe an API for checking the status of the instance?

Reply:
We can use DescribeInstanceStatus API method which is used to check the status of an Instance.

https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceStatus.html